src/modules/instruction-element/instruction-element.service.ts

```diff
@@ -88,4 +118,29 @@ export class InstructionElementService {
 
     await this.instructionElementRepository.delete(id)
   }
+
+  public async getGenericTypes(programId: number): Promise<InstructionElementGenericTypeMap> {
+    const accounts = await this.accountRepository.find({
+      where: {
+        program: {
+          id: programId,
+        },
+      },
+    })
+
+    let customTypes = []
+
+    for (let account of accounts) {
+      customTypes.push({
+        id: account.id,
+        name: account.name,
+        type: InstructionElementGenericType.CUSTOM_ACCOUNT,
+      })
+    }
+
+    let instructionElementGenericTypeMap = new InstructionElementGenericTypeMap
```

Comment:
const instead of let, and pls set up eslint for your IDEA, because a project has these rules!